<launch>
  <!-- Console launch prefix -->
  <arg name="robot_namespace" default="rollo"/>
  <arg name="debug" default="false"/>
  <arg name="launch_prefix" value="gdb -ex run --args" if="$(arg debug)"/>
  <arg name="launch_prefix" value="" unless="$(arg debug)"/>

  <group ns="$(arg robot_namespace)">
    <!-- Config and weights folder. -->
    <arg name="torch_trt_ros"          default="$(find torch_trt_ros)/"/>

    <!-- ROS and network parameter files -->
    <arg name="ros_param_file"             default="$(find torch_trt_ros)/config/rs_all_in_one.yaml"/>
    <arg name="network_param_file"         default="$(find torch_trt_ros)/config/detection_model_rs.yaml"/>

    <!-- Load parameters -->
    <rosparam command="load" ns="torch_trt_ros" file="$(arg ros_param_file)"/>
    <rosparam command="load" ns="torch_trt_ros" file="$(arg network_param_file)"/>

    <!-- Start darknet and ros wrapper -->
   <node name="torch_trt_ros" pkg="torch_trt_ros" type="detect.py" output="screen" launch-prefix="$(arg launch_prefix)">
      <param name="robot_name"            value="$(arg robot_namespace)"/>
    </node>
  </group>
 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->

  <!-- Image decomporessor -->
  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_front"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_left"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

  <include file="$(find sensor_image_utils)/launch/rs_compressed_to_raw.launch">
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="camera_name" value="camera_right"/>
    <arg name="throttle" value="true"/>
    <arg name="image_in" value="image_raw"/>
    <arg name="image_out" value="image_raw"/>
  </include>

</launch>
